.github/workflows/deploy-release.yml

```diff
@@ -0,0 +1,55 @@
+name: Deploy Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Existing tag to be released, e.g. v1.2.3, v2.2.2, v3.0.0'
+        required: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Deploy release
+    steps:
+      - name: Determine tag to be released
+        run: |
+          TAG=${GITHUB_REF:10}
+          if [ -n "${{ github.event.inputs.tag }}" ]; then
+            TAG="${{ github.event.inputs.tag }}" 
+          fi
+          echo "TAG=${TAG}" >> $GITHUB_ENV
+          echo "VERSION=${TAG:1}" >> $GITHUB_ENV
+      - uses: actions/checkout@v4
+        with:
+          ref: "${{ env.TAG }}"
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: release-repository
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+      - name: Deploy release
+        run: ./mvnw deploy -Prelease --batch-mode -DskipTests -DretryFailedDeploymentCount=3
+        env:
+          MAVEN_USERNAME: ${{ secrets.SONATYPE_OSS_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.SONATYPE_OSS_PASSWORD }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+
+      - name: Create GitHub Release
+        uses: softprops/action-gh-release@v2
+        with:
+          # We want to create draft release to be able to edit body before actually releasing it.
```

Comment:
Can we automate what goes into the body? 
Is it releases notes or list of commits?

Reply:
The other plugins has a drafter, which adds a line to the draft release every time a commit is pushed or a pr is merged, single line at a time until it's released.

I preferred to edit the release notes myself, github has "generate release notes from changes from last release feature" in the UI as well, so we don't lose much of that functionality.

Reply:
> Is it releases notes or list of commits?
It's list of prs or commits as the release notes.

Reply:
Nice to know about github has "generate release notes from changes from last release feature" which I wasn't aware of. 
We can add that to wiki instructions

Reply:
Resolving as @srknzl confirmed he will add this to wiki soon